safe/impact_function/multi_exposure_wrapper.py

```diff
@@ -876,6 +912,13 @@ def run(self):
 
             analysis_layers.append(impact_function.analysis_impacted)
             aggregation_layers.append(impact_function.aggregation_summary)
+            list_of_analysis_path.append(
+                impact_function.analysis_impacted.source())
```

Comment:
Using `source`, we don't know which provider it is (postgres, ogr...), can you switch to https://github.com/inasafe/inasafe/blob/develop/safe/gis/tools.py#L132 and the next function to read it?